python/lsst/daf/base/propertyContainer/propertyContainerContinued.py

```diff
@@ -248,64 +264,101 @@ def _guessIntegerType(container, name, value):
         Name of item
 
     value : `object`
-        Value to be assigned a type
+        Value to be assigned a type. Can be an iterable.
 
     Returns
     -------
     useType : `str` or none
         Type to use for the supplied value. `None` if the input is
         `bool` or a non-integral value.
     """
-    useType = None
     maxInt = 2147483647
     minInt = -2147483648
     maxLongLong = 2**63 - 1
     minLongLong = -2**63
     maxU64 = 2**64 - 1
     minU64 = 0
 
-    # We do not want to convert bool to int so let the system work that
-    # out itself
-    if isinstance(value, bool):
-        return useType
-
-    if isinstance(value, numbers.Integral):
-        try:
-            containerType = _propertyContainerElementTypeName(container, name)
-        except LookupError:
-            # nothing in the container so choose based on size.
-            if value <= maxInt and value >= minInt:
-                useType = "Int"
-            elif value <= maxLongLong and value >= minLongLong:
-                useType = "LongLong"
-            elif value <= maxU64 and value >= minU64:
-                useType = "UnsignedLongLong"
-            else:
-                raise RuntimeError("Unable to guess integer type for storing value: %d" % (value,))
+    # Go through the values to find the range of supplied integers,
+    # stopping early if we don't have an integer.
+    min = None
+    max = None
+    for v in _iterable(value):
+        # Do not try to convert a bool to an integer
+        if not isinstance(v, numbers.Integral) or isinstance(v, bool):
+            return None
+
+        if min is None:
+            min = v
+            max = v
+        elif v < min:
+            min = v
+        elif v > max:
+            max = v
+
+    # Safety net
+    if min is None or max is None:
+        raise RuntimeError(f"Internal logic failure calculating integer range of {value}")
```

Comment:
This error message is probably wrong. Can PropertySet store an empty list? If I try it I get StopIteration because that's what `yield from []` is going to do. Implies my calls to `next(_iterable())` are going to have to trap StopIteration.

Reply:
Changed the code above to be a no-op for empty list to match C++ beahvior.